napari/settings/_fields.py

```diff
@@ -82,17 +90,17 @@ def validate(cls, v):
 
 
 @total_ordering
-@dataclass
+@dataclass(eq=False, order=False)
```

Comment:
This should have been the case already, since we provide our own comparison methods. Issues appeared only once we enforce coercion, because pydantic does some weird stuff and dynamically creates dataclasses that are *almost* the same as the field annotation, but in doing so it basically overrides our custom methods unless we explicitly say `False` in the dataclass creation.